packages/fuels-test-helpers/src/node.rs

```diff
@@ -271,4 +298,32 @@ pub async fn server_health_check(client: &FuelClient) {
     if !healthy {
         panic!("error: Could not connect to fuel core server.")
     }
+    true
+}
+
+pub fn get_socket_address() -> SocketAddr {
+    let free_port = pick_unused_port().expect("No ports free");
+    SocketAddr::new("127.0.0.1".parse().unwrap(), free_port)
+}
+
+pub struct FuelService {
+    pub bound_address: SocketAddr,
+}
+
+impl FuelService {
+    pub async fn new_node(config: Config) -> Result<Self, AnyError> {
+        let bound_address = match config {
+            node_config if node_config.addr.port() != 0 && is_free(node_config.addr.port()) => {
```

Comment:
I found this difficult to read, requiring paper to understand.

It reads like this: 
if the port of the address is not zero and is free use it
if it is zero or if it is something else but is not free, check if it is not free, if so panic
otherwise get a random address.

Perhaps a simpler way? 
```rust
        let requested_port = config.addr.port();

        let chosen_addr = if requested_port == 0 {
            get_socket_address()
        } else if is_free(requested_port) {
            config.addr
        } else {
            panic!("Error: Address already in use");
        };
```